### PR TITLE
[v1][torch.compile] support managing cudagraph buffer

### PR DESCRIPTION
**Pull request by youkaichao**
advantage: caller does not need to explicitly manage the input buffer

disadvantage: the cost of one more input tensor, and one more copy kernel.

given the code simplification from user side, i think it should be worthwhile.

### CODE REVIEW

File: tests/compile/piecewise/test_simple.py [hunk 1]
```diff
@@ -94,13 +94,14 @@ def test_simple_piecewise_compile():
...
             model(input_buffer)
 
-            model(input_buffer[:2])
-            model(input_buffer[:1])
+            model(torch.randn(2).cuda())
+            model(torch.randn(1).cuda())
 
-        input_buffer[:2].zero_()
+        input = torch.randn(2).cuda()
+        input.zero_()
```
Comment (WoosukKwon) [inline tests/compile/piecewise/test_simple.py:101]: nit:
```suggestion
        input = torch.zeros(2).cuda()
```

File: tests/compile/piecewise/test_simple.py [hunk 2]
```diff
@@ -94,13 +94,14 @@ def test_simple_piecewise_compile():
         with set_compile_context([1, 2]):
             model(input_buffer)
```
Comment (WoosukKwon) [inline tests/compile/piecewise/test_simple.py:95]: To double check: Here, `input_buffer` is not directly cached and reused by the backend, right?

**Review by WoosukKwon — APPROVED** [review]

File: tests/compile/piecewise/test_simple.py (hunk 2 above)
Comment (youkaichao) [reply tests/compile/piecewise/test_simple.py:95]: right. let me rename the variable. it is not a persistent buffer anymore. it can be a newly created tensor.

File: vllm/compilation/config.py [hunk 3]
```diff
@@ -78,6 +83,7 @@ class CompilationConfig(BaseModel):
     non_cudagraph_ops: List[str] = Field(default_factory=list)
     cudagraph_num_of_warmups: int = 0
     cudagraph_capture_sizes: Optional[List[int]] = None
+    cudagraph_copy_inputs: bool = True
```
Comment (WoosukKwon) [inline vllm/compilation/config.py:86]: What about making it False by default?
